Compile error on Windows:  error C2039: '_copysign': is not a member of 'boost::core'

Bug Report
==========
<!--
Please fill in a a meaningful and concise title in the field above.
It helps us to classify the issue even  before reading the body and handling the
issue a bit better.

Examples:
  * Great: "Unable to select fleet in galaxy map"
  * Good: "Unable to click fleet"
  * Not really okay: "Issue with fleets"
  * Horrible: "HELP!!!"
-->

Environment
-----------
<!--
You don't need to provide the information in section when they are not useful in
the context of this issue.  For example a crashing hosted server may not need
information about the graphic card, but maybe depend on the operating system you
use.  When in doubt provide all information you have available.

You can find the FreeOrion version number in the lower right corner of the game
main screen.

When stating the operating system also note the version of the operating system
so, `Windows 8 Pro` instead of `Windows` or `Mac OSX Mountain Lion` instead of
`OSX`.
-->

* **FreeOrion Version**: eb6a7d214cc74e3fdfe2b16d31ab3c011e59b0b9
* **Operating System**: Windows 10
* **Fetched as** <!-- delete lines below that don't apply -->
  * Compiled from source


Description
-----------
<!--
Add a meaningful description of the bug you encountered.

If a screenshot or image helps to describe the issue content feel free to create
one.

Also attach the log files the game creates.  The log files The log files can be
found on
* Windows: %APPDATA%\FreeOrion
* MacOSX: ~/Library/"Application Support"/FreeOrion
* Linux: ${XDG_DATA_HOME:-~/.local/share}/freeorion
-->
Trying to build FO on Windows fails with this error message:
```
1>C:\Users\User\SoftwareProjekte\FO\main\include\boost\lexical_cast\detail\inf_nan.hpp(80,43): error C2039: '_copysign': is not a member of 'boost::core'
1>C:\Users\User\SoftwareProjekte\FO\main\include\boost\core\is_same.hpp(23): message : see declaration of 'boost::core'
```
As required, I'm using v13 of the SDK to build master at this commit.

--- a/.github/workflows/build-windows-msvs.yml
+++ b/.github/workflows/build-windows-msvs.yml
@@ -1,0 +1,49 @@
+name: "Windows MS VS build"
+
+on:
+  push:
+    paths-ignore:
+      - 'check/**'
+      - 'doc/**'
+      - 'packaging/**'
+      - 'snap/**'
+      - '*.md'
+      - 'check/**'
+      - 'default/**'
+      - 'test-scripting/**'
+  pull_request:
+    paths-ignore:
+      - 'check/**'
+      - 'doc/**'
+      - 'packaging/**'
+      - 'snap/**'
+      - '*.md'
+      - 'check/**'
+      - 'default/**'
+      - 'test-scripting/**'
+
+jobs:
+  build-windows-2022:
+    name: MSVS 2022 on Windows
+    runs-on: windows-2022
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - name: Download SDK
+        id: download-sdk
+        uses: suisei-cn/actions-download-file@v1
+        with:
+          url: https://github.com/freeorion/freeorion-sdk/releases/download/v13/FreeOrionSDK_13_MSVC-v141-Win32.zip
+          target: ../
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.0.3
+        with:
+            vs-prerelease: true
+      - name: Prepare
+        run: |
+          unzip -q ../${{ steps.download-sdk.outputs.filename }} -d ..
+          cp ../bin/* ./
+      - name: Build
+        run: |
+          cd msvc2022
+          msbuild FreeOrion.sln /maxcpucount /property:BuildInParallel=true /property:CL_MPCount=2 /property:PlatformToolset=v142 /property:Configuration=Release /verbosity:minimal

--- a/.github/workflows/build-windows.yml
+++ b/.github/workflows/build-windows.yml
@@ -26,7 +26,7 @@ on:
 
 jobs:
   build-windows-2019-cmake:
-    name: MSVS 2019 on Windows
+    name: MSVS 2019 on Windows CMake
     runs-on: windows-2019
     steps:
       - name: Checkout sources
@@ -93,7 +93,7 @@ jobs:
           retention-days: 14
 
   build-windows-2022-cmake:
-    name: MSVS 2022 on Windows 
+    name: MSVS 2022 on Windows CMake
     runs-on: windows-2022
     steps:
       - name: Checkout sources

--- a/msvc2022/Parsers/Parsers.vcxproj
+++ b/msvc2022/Parsers/Parsers.vcxproj
@@ -120,7 +120,7 @@
       <AdditionalOptions>
       </AdditionalOptions>
       <ForcedIncludeFiles>StdAfx.h;%(ForcedIncludeFiles)</ForcedIncludeFiles>
-      <LanguageStandard>stdcpp20</LanguageStandard>
+      <LanguageStandard>stdcpp17</LanguageStandard>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>
@@ -152,7 +152,7 @@
       <FavorSizeOrSpeed>Size</FavorSizeOrSpeed>
       <AdditionalOptions>/MP4 %(AdditionalOptions)</AdditionalOptions>
       <ForcedIncludeFiles>StdAfx.h;%(ForcedIncludeFiles)</ForcedIncludeFiles>
-      <LanguageStandard>stdcpp20</LanguageStandard>
+      <LanguageStandard>stdcpp17</LanguageStandard>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>

